packages/component-base/test/slot-observer.test.js

```diff
@@ -18,10 +19,12 @@ describe('SlotObserver', () => {
     host.remove();
   });
 
-  it('should run callback for initial nodes synchronously', () => {
+  it('should run callback for initial nodes asynchronously', async () => {
     spy = sinon.spy();
     observer = new SlotObserver(slot, spy);
-    expect(spy.called).to.be.true;
+    expect(spy.called).to.be.false;
+
+    await aTimeout(0);
```

Comment:
Should we wait for `Promise.resolve()` in order to assert the timing more accurately?

Reply:
In this case it's not enough because we have `slotchange` event fired at microtask, plus another microtask created by `_schedule()` call. Updated the tests accordingly and added comments to clarify this.